### ISSUE TITLE
Installation fails on termux Python 3.11 

### ISSUE THREAD

**Issue by not-lum** [numba/numba#9685]
## Reporting a bug

- [x] I have tried using the latest released version of Numba (most recent is
  visible in the release notes
  (https://numba.readthedocs.io/en/stable/release-notes-overview.html).
- [x] I have included a self contained code sample to reproduce the problem.
  i.e. it's possible to run as 'python bug.py'.

### Steps to reproduce (run on termux):
`python3.11 -m pip install numba`

### Description:
Installation on termux on python 3.11 fails with the following error:
```c
      aarch64-linux-android-clang -DNDEBUG -g -fwrapv -O3 -Wall -fstack-protector-strong -O3 -fstack-protector-strong -O3 -fPIC -I/data/data/com.termux/files/usr/tmp/pip-build-env-xsazcb5s/normal/lib/python3.11/site-packages/numpy/_core/include -I/data/data/com.termux/files/usr/include/python3.11 -c numba/_helpermod.c -o build/temp.linux-aarch64-cpython-311/numba/_helpermod.o
      In file included from numba/_helpermod.c:23:
      numba/_helperlib.c:147:12: error: use of undeclared identifier 'I'
        147 |     *out = _complex_float_ctor((float) _out.real, (float) _out.imag);
            |            ^
      numba/_helperlib.c:22:44: note: expanded from macro '_complex_float_ctor'
         22 |     #define _complex_float_ctor(r, i) (r + I * i)
            |                                            ^
      1 error generated.
```
So, this line is the cause of this error:
https://github.com/numba/numba/blob/2001717f3321a5082c39c5787676320e699aed12/numba/_helperlib.c#L22
I may be wrong, but it seems to me that there is a typo and it should be like this:
```c
 #define _complex_float_ctor(r, i) (r + i * i)
```

**Comment by tiran** [comment]
The problem is caused by a recent change in NumPy 2.0.1, https://github.com/numpy/numpy/issues/27083

**Comment by randyli** [comment]
Downgrade the numpy maybe works 
```
pip install numpy==1.22
```

**Comment by gmarkall** [comment]
Addressed by #9683.
